tests/network/sriov.go

```diff
@@ -520,6 +523,42 @@ var _ = Describe("[Serial]SRIOV", func() {
 			})
 		})
 
+		Context("Connected to multiple SRIOV networks", func() {
+			sriovNetworks := []string{sriovnet1, sriovnet2, sriovnet3, sriovnet4}
+			BeforeEach(func() {
+				for _, sriovNetwork := range sriovNetworks {
+					Expect(createSriovNetworkAttachmentDefinition(sriovNetwork, util.NamespaceTestDefault, sriovConfNAD)).To(Succeed(), shouldCreateNetwork)
+				}
+			})
+
+			It("should correctly plug all the interfaces based on the specified MAC and (guest) PCI addresses", func() {
+				macAddressTemplate := "de:ad:00:be:ef:%02d"
+				pciAddressTemplate := "0000:%02d:00.0"
+				vmi := getSriovVmi(sriovNetworks, defaultCloudInitNetworkData())
+				for i := range sriovNetworks {
+					vmi.Spec.Domain.Devices.Interfaces[i+1].PciAddress = fmt.Sprintf(pciAddressTemplate, i)
+					vmi.Spec.Domain.Devices.Interfaces[i+1].MacAddress = fmt.Sprintf(macAddressTemplate, i)
+				}
+
+				vmi = startVmi(vmi)
+				vmi = waitVmi(vmi)
+
+				Eventually(func() error {
+					for _, iface := range vmi.Spec.Domain.Devices.Interfaces {
+						guestInterfaceName, err := getInterfaceNameByMAC(vmi, iface.MacAddress)
+						Expect(err).ToNot(HaveOccurred())
+						if len(guestInterfaceName) == 0 {
+							return fmt.Errorf("guest InterfaceName not updated in iface %s", iface.Name)
```

Comment:
I think that the way errors are handled inside the Eventually func is confusing:
line 549 the error is asserted (and blow up the test).
line 551 the error is not asserted (returned instead).
Is it possible to handle both (and other) errors the same way?

Reply:
I suggest to take a look please on
```
					ifaceName, err := findIfaceByMAC(virtClient, vmi, mac, 30*time.Second)
					Expect(err).NotTo(HaveOccurred())
```
It has the Eventually in it already and then this block won't be needed
(at least some of it)

Reply:
couldn't agree more. thanks!